### ISSUE TITLE
How to use STDC2 and custom dataset?

### ISSUE THREAD

**Issue by duran67** [zh320/realtime-semantic-segmentation-pytorch#25]
Hi and thanks for a great project.

I have two questions

1. In PPLiteSeg and STDC you have encoder_type='stdc1', how can I specify 'stdc2'?

2. Is it possible to add a custom dataset to datasets/ and if so, how do I do that?

using revision b4c9a39 (Mar 10, 2025)

**Comment by zh320** [comment]
Hi,

1. You need to change `encoder_type='stdc2'` within the following lines

https://github.com/zh320/realtime-semantic-segmentation-pytorch/blob/f67dba4959a8074c9162eeb5cb9af7e231ab0d16/models/pp_liteseg.py#L19

https://github.com/zh320/realtime-semantic-segmentation-pytorch/blob/f67dba4959a8074c9162eeb5cb9af7e231ab0d16/models/stdc.py#L19

2. Yes, it is relatively easy to create a custom dataset. You may code whatever data flow you would like, but need to make sure
- the dataset class should return image and the segmentation mask, and both are torch.Tensor
- the image should be normalized in general
- the mask id should be continuous, except the one if you use `ignore_index` for the loss.
- the training pipeline does not provide augmentation workflow. I would suggest implementing the augmentation workflow within the dataset class as the `Cityscapes` class.
- don't forget to `@register_dataset` your dataset and import it within `datasets/__init__.py`
